src/compiler/checker.ts

```diff
@@ -10506,6 +10913,31 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 return localName;
             }
         }
+
+        function isExpanding(context: NodeBuilderContext) {
+            return context.maxExpansionDepth !== -1;
+        }
+
+        function isHashPrivate(s: Symbol): boolean {
+            // `valueDeclaration` could be undefined if inherited from
+            // a union/intersection base type, but inherited properties
+            // don't matter here.
+            return !!s.valueDeclaration && isNamedDeclaration(s.valueDeclaration) && isPrivateIdentifier(s.valueDeclaration.name);
+        }
+
+        function getHashPrivateName(s: Symbol): PrivateIdentifier | undefined {

```

Comment:
nit: The name of this method should probably indicate it returns a clone of the name, not the original.